presto-main/src/main/java/com/facebook/presto/operator/Driver.java

```diff
@@ -273,11 +274,17 @@ private void processNewSources()
             if (fragmentResultCacheContext.get().isPresent() && !(split.getConnectorSplit() instanceof RemoteSplit)) {
                 checkState(!this.cachedResult.get().isPresent());
                 this.fragmentResultCacheContext.set(this.fragmentResultCacheContext.get().map(context -> context.updateRuntimeInformation(split.getConnectorSplit())));
-                Optional<Iterator<Page>> pages = fragmentResultCacheContext.get().get()
+                FragmentCacheResult fragmentCacheResult = fragmentResultCacheContext.get().get()
                         .getFragmentResultCacheManager()
                         .get(fragmentResultCacheContext.get().get().getHashedCanonicalPlanFragment(), split);
+                Optional<Iterator<Page>> pages = fragmentCacheResult.getPages();
                 sourceOperator.getOperatorContext().getRuntimeStats().addMetricValue(
                         pages.isPresent() ? FRAGMENT_RESULT_CACHE_HIT : FRAGMENT_RESULT_CACHE_MISS, NONE, 1);
+                if (pages.isPresent()) {
+                    sourceOperator.getOperatorContext().getRuntimeStats().addMetricValue("FRAGMENT_DATA_SIZE", BYTE, fragmentCacheResult.getInputDataSize());
```

Comment:
does this give the correct FRC cache size?

Reply:
yes it is . confirmed by logging.